modules/siddhi-core/src/main/java/io/siddhi/core/util/cache/CacheExpiryHandler.java

```diff
@@ -0,0 +1,225 @@
+        metaStateEvent.addEvent(currentTimeMetaStreamEvent);
+        metaStateEvent.addEvent(tableMetaStreamEvent);
+
+        MatchingMetaInfoHolder matchingMetaInfoHolder =
+                MatcherParser.constructMatchingMetaStateHolder(metaStateEvent, 0,
+                        cacheTable.getTableDefinition(), 0);
+        List<VariableExpressionExecutor> variableExpressionExecutors = new ArrayList<>();
+        Expression leftExpressionForSubtract = new Variable(CACHE_EXPIRE_CURRENT_TIME);
+        Expression leftExpressionForCompare = new Subtract(leftExpressionForSubtract, rightExpressionForSubtract);
+        Expression deleteCondition = new Compare(leftExpressionForCompare, greaterThanOperator,
+                rightExpressionForCompare);
+        SiddhiQueryContext siddhiQueryContext = new SiddhiQueryContext(siddhiAppContext, "expiryDeleteQuery");
+        return cacheTable.compileCondition(deleteCondition, matchingMetaInfoHolder, variableExpressionExecutors,
+                tableMap, siddhiQueryContext);
+    }
+
+    private void handleCacheExpiry() {
+        log.debug(siddhiAppContext.getName() + ": CacheExpiryHandler started");
+        StateEvent stateEventForCaching = new StateEvent(1, 0);
+        StreamEvent loadedDataFromStore;
+
+        if (storeTable.getStoreTableSize() != -1 && storeTable.getStoreSizeLastCheckedTime() >
+                        siddhiAppContext.getTimestampGenerator().currentTime() - retentionPeriod * 10) {
+            if (log.isDebugEnabled()) {
+                log.debug(siddhiAppContext.getName() + ": checking size of store table");
+            }
+            try {
+            if (storeTable.getStoreTableSize() <= storeTable.getMaxCacheSize()) {
+                AbstractQueryableRecordTable.queryFromStore.set(Boolean.TRUE);
+                try {
+                    loadedDataFromStore = storeTable.query(stateEventForCaching,
+                            storeTable.getCompiledConditionForCaching(),
+                            storeTable.getCompiledSelectionForCaching(), storeTable.getOutputAttributesForCaching());
+                    if (storeTable.getCacheLastReloadTime() < siddhiAppContext.getTimestampGenerator().currentTime() +
+                            retentionPeriod) {
+                        clearCacheAndReload(loadedDataFromStore);
+                        storeTable.setCacheLastReloadTime(siddhiAppContext.getTimestampGenerator().currentTime());
+                    }
+                } finally {
+                    AbstractQueryableRecordTable.queryFromStore.set(Boolean.FALSE);
+                }
+            } else {
+                CompiledCondition cc = cacheExpiryCompiledCondition;
+                cacheTable.delete(generateDeleteEventChunk(),
+                        cc);
+            }
+            } catch (ConnectionUnavailableException e) {
+                throw new SiddhiAppRuntimeException(siddhiAppContext.getName() + ": " + e.getMessage());
```

Comment:
add query name

Reply:
cache expirer operation is independent of query